src/plugins/intel_cpu/src/nodes/interaction.cpp

```diff
@@ -0,0 +1,229 @@
+    } else if (dataPrecision == InferenceEngine::Precision::BF16) {
+        run<int16_t>(strm);
+    }
+}
+
+bool Interaction::created() const {
+    return getType() == Type::Interaction;
+}
+
+void Interaction::prepareParams() {
+    using tag = dnnl::memory::format_tag;
+    using dt = dnnl::memory::data_type;
+    using namespace dnnl;
+    const auto& denseFeatureDims = getParentEdgeAt(0)->getMemory().getStaticDims();
+    batchSize = denseFeatureDims[0];
+    featureSize = denseFeatureDims[1];
+    inputSizes = inputShapes.size();
+    interactFeatureSize = inputSizes * (inputSizes - 1) / 2;
+    outputFeaturesLen = interactFeatureSize + featureSize;
+    std::vector<int64_t> lhsShape({static_cast<int64_t>(inputSizes), static_cast<int64_t>(featureSize)});
+    std::vector<int64_t> lhsStride({static_cast<int64_t>(featureSize), 1});
+    std::vector<int64_t> rhsShape({static_cast<int64_t>(featureSize), static_cast<int64_t>(inputSizes)});
+    std::vector<int64_t> rhsStride({1, static_cast<int64_t>(featureSize)});
+    std::vector<int64_t> resShape({static_cast<int64_t>(inputSizes), static_cast<int64_t>(inputSizes)});
+    std::vector<int64_t> resStride({static_cast<int64_t>(inputSizes), 1});
+    auto dataType = DnnlExtensionUtils::IEPrecisionToDataType(dataPrecision);
+    auto src_md = memory::desc(lhsShape, dataType, lhsStride);
+    auto weights_md = memory::desc(rhsShape, dataType, rhsStride);
+    auto dst_md = memory::desc(resShape, dataType, resStride);
+    auto matmul_d = matmul::desc(src_md, weights_md, dst_md);
+    primitive_attr matmul_attr;
+    auto matmul_pd = matmul::primitive_desc(matmul_d, matmul_attr, getEngine());
+    prim.reset(new matmul(matmul_pd));
+    featureSizes.resize(inputSizes, featureSize);
+    std::vector<InferenceEngine::TensorDesc> internalMemDesc = {
+        InferenceEngine::TensorDesc(
+            dataPrecision,
+            {inputSizes, featureSize},
+            InferenceEngine::Layout::HW),
+        InferenceEngine::TensorDesc(
+            dataPrecision,
+            {inputShapes.size(), inputShapes.size()},
+            InferenceEngine::Layout::HW),
+        InferenceEngine::TensorDesc(
+            dataPrecision,
+            {interactFeatureSize},
+            InferenceEngine::Layout::ANY)
+    };
+
+    if (dataPrecision == InferenceEngine::Precision::FP32) {
+        initializeInternalMemory<float>(internalMemDesc);
+    } else {
+        initializeInternalMemory<int16_t>(internalMemDesc);
+    }
+
+    inputMemPtr = std::make_shared<Memory>(getEngine());
+    outputMemPtr = std::make_shared<Memory>(getEngine());
+    auto inDesc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(inputPtr->getTensorDesc());
+    auto outDesc = MemoryDescUtils::convertToDnnlBlockedMemoryDesc(outputPtr->getTensorDesc());
+    inputMemPtr->Create(inDesc, inputPtr->buffer());
+    outputMemPtr->Create(outDesc, outputPtr->buffer());
+}
+
+void Interaction::executeDynamicImpl(dnnl::stream strm) {
```

Comment:
Add this node to *[make_shape_inference](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/utils/shape_inference/shape_inference.cpp#L406)*

Reply:
Interaction is an internal cpu ops,  `shape_infer` library is implemented inside another library, while it could not access the cpu internal ops. I check a similar op `FullyConnected` layer, it also doesn't use `shape_infer` now, it simply fallbacks to `validate_and_infer_types`. Do you think I need to put shape_infer function of `Interaction` inside cpu codes rather than in shape infer library ?

Reply:
Ok, please add tests for dynamic shapes to make sure that it works correctly.

Reply:
Sure, add dynamic shape tests in commit c67fb3cc606836770a72b723ec7843f237c2864a.